File: tests_deps/src/main/java/com/github/klee0kai/test/car/model/Car.java
package com.github.klee0kai.test.car.model;

import java.util.Collections;
import java.util.List;
import java.util.UUID;

public class Car {

    public static int createCount = 0;
    public String uuid = UUID.randomUUID().toString();

    public List<Bumper> bumpers;
    public List<Wheel> wheels;
    public List<Window> windows;

    public Car(
            Bumper bumper,
            Wheel wheel,
            Window window
    ) {
        createCount++;
        bumpers = Collections.singletonList(bumper);
        wheels = Collections.singletonList(wheel);
        windows = Collections.singletonList(window);
    }


    public Car(List<Bumper> bumpers, List<Wheel> wheels, List<Window> windows) {
        createCount++;
        this.bumpers = bumpers;
        this.wheels = wheels;
        this.windows = windows;
    }
}

File: tests_wraps_kotlin/src/main/kotlin/com/github/klee0kai/test/car/model/Car.kt
package com.github.klee0kai.test.car.model

import java.util.*

class Car {
    var uuid = UUID.randomUUID().toString()

    var bumpers: List<Bumper?>?
    var wheels: List<Wheel?>?
    var windows: List<Window?>?
    var qualifier: String? = null

    constructor(
        bumper: Bumper?,
        wheel: Wheel?,
        window: Window?
    ) {
        createCount++
        bumpers = listOf(bumper)
        wheels = listOf(wheel)
        windows = listOf(window)
    }

    constructor(bumpers: List<Bumper?>?, wheels: List<Wheel?>?, windows: List<Window?>?) {
        createCount++
        this.bumpers = bumpers
        this.wheels = wheels
        this.windows = windows
    }

    companion object {
        var createCount = 0
    }
}
